### ISSUE TITLE
Obfuscated access to the IDFA

### ISSUE THREAD

**Issue by ignazioc** [adjust/ios_sdk#302]
Hi,
I'm looking at the code and I'm wondering why the access to the IDFA is obfuscated.
For example here:
``` 
NSString *className = [NSString adjJoin:@"A", @"S", @"identifier", @"manager", nil];
    Class class = NSClassFromString(className);
    if (class == nil) {
        return NO;
    }
``` 
or here:
``` 
NSString *keyEnabled = [NSString adjJoin:@"is", @"advertising", @"tracking", @"enabled", nil];
    SEL selEnabled = NSSelectorFromString(keyEnabled);
    if (![manager respondsToSelector:selEnabled]) {
        return NO;
    }
    BOOL enabled = (BOOL)[manager performSelector:selEnabled];
    return enabled;
```

**Comment by uerceg** [comment]
Hi @ignazioc 

Reason behind using reflection approach to read this is that we didn't want to directly reference classes from `AdSupport.framework` and thus cause compile time errors for people that use our SDK and chose not to add that framework to their app for any reason.

**Comment by ignazioc** [comment]
No compile errors are possible because the code is between:
```
#if ADJUST_NO_IDFA
    //
#else
  //
#endif
```
To me, looks like a way to bypass the Apple check when people forgot to mention that the apple uses the `IDFA`.

**Comment by uerceg** [comment]
@ignazioc 

It wasn't always like this. This code is some of the oldest code we wrote into the SDK and the flag itself was added at some later point in time. For example, this is how it looked back in v2.0: https://github.com/adjust/ios_sdk/blob/v2.0/AdjustIo/AIAdditions/UIDevice%2BAIAdditions.m

Reflection itself is not being used for Apple checks bypassing. We saw that apps who were not serving ads and not ticking the field were being rejected for unclear reasons with class being directly referenced. So, as we said, we are using the reflection approach so that people don't make mistakes during submission, or get rejected without a clear reason.
